shard-operator/src/main/java/com/redhat/service/smartevents/shard/operator/controllers/BridgeIngressController.java

```diff
@@ -85,61 +100,116 @@ public List<EventSource> prepareEventSources(EventSourceContext<BridgeIngress> e
 
     @Override
     public UpdateControl<BridgeIngress> reconcile(BridgeIngress bridgeIngress, Context context) {
-        LOGGER.info("Create or update BridgeIngress: '{}' in namespace '{}'", bridgeIngress.getMetadata().getName(), bridgeIngress.getMetadata().getNamespace());
+        LOGGER.info("Create or update BridgeIngress: '{}' in namespace '{}'",
+                bridgeIngress.getMetadata().getName(),
+                bridgeIngress.getMetadata().getNamespace());
+
+        BridgeIngressStatus status = bridgeIngress.getStatus();
+
+        if (isTimedOut(status)) {
+            notifyManagerOfFailure(bridgeIngress,
+                    new ProvisioningTimeOutException(String.format(ProvisioningTimeOutException.TIMEOUT_FAILURE_MESSAGE,
+                            bridgeIngress.getClass().getSimpleName(),
+                            bridgeIngress.getSpec().getId())));
+            status.markConditionFalse(ConditionTypeConstants.READY);
+            status.markConditionFalse(BridgeIngressStatus.SECRET_AVAILABLE);
+            status.markConditionFalse(BridgeIngressStatus.CONFIG_MAP_AVAILABLE);
+            status.markConditionFalse(BridgeIngressStatus.KNATIVE_BROKER_AVAILABLE);
+            status.markConditionFalse(BridgeIngressStatus.AUTHORISATION_POLICY_AVAILABLE);
+            status.markConditionFalse(BridgeIngressStatus.NETWORK_RESOURCE_AVAILABLE);
+            return UpdateControl.updateStatus(bridgeIngress);
+        }
 
         Secret secret = bridgeIngressService.fetchBridgeIngressSecret(bridgeIngress);
 
         if (secret == null) {
             LOGGER.info("Secrets for the BridgeIngress '{}' have been not created yet.", bridgeIngress.getMetadata().getName());
-            bridgeIngress.getStatus().markConditionFalse(ConditionTypeConstants.READY);
-            bridgeIngress.getStatus().markConditionTrue(ConditionTypeConstants.AUGMENTATION, ConditionReasonConstants.SECRETS_NOT_FOUND);
-            return UpdateControl.updateStatus(bridgeIngress);
+            if (!status.isConditionTypeFalse(ConditionTypeConstants.READY)) {
```

Comment:
I only update the `Condition` if it has not already been done so otherwise the "last transitioned date" is updated each time the reconcile loop iterates.